src/DeviceInterface/ExtensionInterface.js

```diff
@@ -10,17 +10,19 @@
 } from '../autofill-utils.js'
 import {HTMLTooltipUIController} from '../UI/controllers/HTMLTooltipUIController.js'
 import {defaultOptions} from '../UI/HTMLTooltip.js'
-import {
-    SetIncontextSignupInitiallyDismissedAtCall,
-    SetIncontextSignupPermanentlyDismissedAtCall
-} from '../deviceApiCalls/__generated__/deviceApiCalls.js'
+import {InContextSignup} from "../InContextSignup";
 
 const TOOLTIP_TYPES = {
     EmailProtection: 'EmailProtection',
     EmailSignup: 'EmailSignup'
 }
 
 class ExtensionInterface extends InterfacePrototype {
+    /**
+     * Adding this here since only the extension currently supports this
+     */
+    inContextSignup = new InContextSignup(this)
```

Comment:
we can prevent the base `InterfacePrototype` from growing in size for every feature we add by using composition like this and dependency injection so that the feature can call back to the device.